src/synthesizer/parametric/morphology.py

```diff
@@ -46,30 +51,64 @@ class Sersic2D(MorphologyBase):
     """
     A class holding the Sersic2D profile. This is a wrapper around the
     astropy.models.Sersic2D class.
-
-    Methods
-    -------
-    compute_density_grid
-        Calculates and returns the density grid defined by this Morphology
     """
 
     def __init__(
         self,
-        r_eff_kpc=None,
+        r_eff=None,
         r_eff_mas=None,
-        n=1,
-        ellip=0,
+        r_eff_kpc=None,
+        sersic_index=1,
+        ellipticity=0,
         theta=0.0,
         cosmo=None,
         redshift=None,
     ):
-        """ """
+        """
+        Initialise the morphology.
+        
+        Arguments
+            r_eff (unyt)
+                Effective radius. This is converted as required.
+            sersic_index (float)
+                Sersic index.
+            ellipticity (float)
+                Ellipticity.
+            theta (float)
+                Theta, the rotation angle.
+            cosmo (astro.cosmology)
+                astropy cosmology object.
+            redshift (float)
+                Redshift.
+
+        """
+        self.r_eff_mas = None
+        self.r_eff_kpc = None
+
+        # Check units of r_eff and convert if necessary.
+        if isinstance(r_eff, unyt_array):
+            if r_eff.units.dimensions == length:
+                self.r_eff_kpc = r_eff.to('kpc').value
+            elif r_eff.units.dimensions == angle:
+                self.r_eff_mas = r_eff.to('mas').value
+            else:
+                raise exceptions.IncorrectUnits(
+                    "The units of r_eff must have length or angle dimensions"
+                    )
+            self.r_eff = r_eff
+        elif r_eff_mas:
```

Comment:
I see that here the argument is used. Still, I think the singular argument is good enough.